### ISSUE TITLE
Feature request: nemo actions submenus

### ISSUE THREAD

**Issue by Frunkh** [linuxmint/nemo#3228]
```
 * Nemo version 4.4.2
 * Is issue with desktop and windowed nemo
 * Distribution - Zorin OS 16.2
 * Graphics hardware *and* driver used NVIDIA 1080ti. Driver 525
 * 64 bit
 ```

**Issue**
When defining custom nemo actions, I am unable to provide a parent menu item name.
I can only make individual menu items but I'd love to group certain menu items under a parent menu.

**Comment by amiga-500** [comment]
Yea ive been wondering the same for years.  Seems like an oversight not to allow custom submenus.

**Comment by devingfx** [comment]
Yep! +1

Nemo actions are so simple and practical that you end up to have hundreds of them, grouping is a must have!

**Comment by ramuyk** [comment]
I completely agree that submenus for Nemo actions should be a must-have... In my perspective, Nemo is much more responsive and faster than Dolphin in multiple aspects... But I end up sticking with Dolphin because after having 30 actions we can't handle them properly without submenus...

**Comment by Jeremy7701** [comment]
Very Useful - but I can see that submenus would be desirable (although it might be possible to restrict some actions according to file type).

**Comment by devingfx** [comment]
@Jeremy7701 Here are my use cases:
As I'm experimenting with AI image tools like Stable Diffusion, I use nemo-actions to send image(s) to several APIs, it would be nice to tidy up the image file context menu (sorted randomly):

- Get PNG info
- Get BLIP captiioning
- Get CLIP interrogator
- Get DeepDanbooru prompt
- Send to img2img
- Send to inpaint
- Send to outpaint
- Send to upscale 2x
- Send to upscale 4x
- Remove background
- Get background mask



It would be good to have:

- Get metadatas >
    - PNG info
    - BLIP captiioning
    - CLIP interrogator
    - DeepDanbooru prompt
- Send to SD >
    - img2img
    - inpaint
    - outpaint
    - upscale >
        - 2x
        - 4x
- Background >
    - Remove
    - Get mask

**Comment by BloodyRain2k** [comment]
As someone who recently started looking into Actions and Scripts, I am too very confused that there's no option for grouping. I even wanted to try scripts instead, but for those I can't even find an instruction on how to create them...
I really wish that there would be the option for custom submenu groups, because I already know at this point that I'll need them.

**Comment by Jeremy7701** [comment]
Scripts have some help text for this:-
Navigate to `~/.local/share/nemo/scripts/` and click on the 'more info' button;

![Screenshot from 2023-12-30 12-52-41](https://github.com/linuxmint/nemo/assets/52585119/3e55ce80-e09c-49e2-b655-c4288008382d)
This should display
```
Scripts
When executed from a local folder, scripts will be passed the selected file names.
When executed from a remote folder (e.g. a folder showing web or ftp content),
scripts will be passed no parameters.

In all cases, the following environment variables will be set by Nemo, which the scripts may use:

NEMO_SCRIPT_SELECTED_FILE_PATHS: newline-delimited paths for selected files (only if local)
NEMO_SCRIPT_SELECTED_URIS: newline-delimited URIs for selected files
NEMO_SCRIPT_CURRENT_URI: URI for current location
NEMO_SCRIPT_WINDOW_GEOMETRY: position and size of current window
NEMO_SCRIPT_NEXT_PANE_SELECTED_FILE_PATHS: newline-delimited paths for selected files
 in the inactive pane of a split-view window (only if local)
NEMO_SCRIPT_NEXT_PANE_SELECTED_URIS: newline-delimited URIs for selected files
 in the inactive pane of a split-view window
NEMO_SCRIPT_NEXT_PANE_CURRENT_URI: URI for current location in the inactive pane of a split-view window
```

**Comment by BloodyRain2k** [comment]
Yes I did read that text. But the fact that I added two scripts and the submenu entry `Scripts` didn't even show up until half a day later, because I needed to restarted Nemo due to having severely broken something somehow, threw me off. Because the `Actions` I created showed up immediately.

But even then, there's the problem that it's apparently not possible to give the scripts conditions when they should show up and when not.
Or if there is, then the note doesn't point it out. Which is why I'd prefer a sample script there too, like there's a sample action. Even if it's just a shell script that echo's the filenames to a textfile.

**Comment by Jeremy7701** [comment]
I suspect you would need to restart nemo to enable scripts - but scripts seem aimed at multiple files whilst actions seem aimed at single files and allow you to restrict them by mime type or "extension".

I found an example of a simple script at [https://itsfoss.com/nemo-tweaks/](https://itsfoss.com/nemo-tweaks/) which reads:
```
#!/bin/bash
name=$(zenity --color-selection --show-palette --title Color\ Select)
copyq add $name
```

**Comment by LiveFreeDead** [comment]
I'd also love to have this feature added, it would be very nice to have a few sub menu's that still only show if populated with a menu item (correct mimetype or extensions/dir etc), I'd have one for Extract Tools, Picture Tools, System Tools. Yeah you can make a dialog that gives each option, but it's about simplicity, it's something modern OS's offer, so it's sad it's not available to us.

**Comment by mtwebster** [comment]
I must have missed this issue when closing them... this has been implemented in nemo 6.2:

![image](https://github.com/user-attachments/assets/d36ea730-ec24-4cda-a4fd-6df82bf2cee5)
